src/frontend/citizen-portal/src/app/components/initiate-resolution/initiate-resolution.component.ts

```diff
@@ -13,26 +13,30 @@ import { ViolationTicket } from "app/api";
   styleUrls: ["./initiate-resolution.component.scss"],
 })
 export class InitiateResolutionComponent implements OnInit {
+  private params: any;
   public busy: Subscription;
   public ticket: ViolationTicket;
   public ticketType: string;
   public ticketTypes = ticketTypes;
 
   constructor(
+    protected route: ActivatedRoute,
     protected router: Router,
     private logger: LoggerService,
     private violationTicketService: ViolationTicketService,
   ) {
     // always reconstruct current component
     this.router.routeReuseStrategy.shouldReuseRoute = () => { return false; };
+    this.route.queryParams.subscribe((params) => {
+      this.params = params;
+    });
   }
 
   public ngOnInit(): void {
-    if (this.violationTicketService.validateTicket()) {
+    if (this.violationTicketService.validateTicket(this.params)) {
```

Comment:
Is there a reason why this cant be strongly typed?

Reply:
It can be, will modify it later